Build errors in release in terminal\src\tools\closetest\closetest.cpp

### Windows Terminal version

n/a

### Windows build number

10.0.19044.1620

### Other Software

This s a repo synced to:

C:\Users\micha\source\repos\terminal>git show
commit f5882236dcb8693a0b14c70e4bbe73e6401d844d (HEAD -> main, origin/main, origin/HEAD)
Author: Dan Mezhiborsky <danielmarkm@gmail.com>
Date:   Thu Mar 24 17:08:57 2022 -0400


### Steps to reproduce

Sync to:

C:\Users\micha\source\repos\terminal>git show
commit f5882236dcb8693a0b14c70e4bbe73e6401d844d (HEAD -> main, origin/main, origin/HEAD)
Author: Dan Mezhiborsky <danielmarkm@gmail.com>
Date:   Thu Mar 24 17:08:57 2022 -0400


tools\razzle rel
bcz

...


       "C:\Users\micha\source\repos\terminal\OpenConsole.sln" (Clean;Build target) (1) ->
       "C:\Users\micha\source\repos\terminal\src\host\ft_uia\Host.Tests.UIA.csproj.metaproj" (Clean;Build target) (18)
       ->
       "C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.vcxproj" (default target) (42:3) ->
       (ClCompile target) ->
         C:\Users\micha\source\repos\terminal\src\tools\closetest\closetest.cpp(301,17): error C4189: 'count2': local v
       ariable is initialized but not referenced [C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.vc
       xproj]
         C:\Users\micha\source\repos\terminal\src\tools\closetest\closetest.cpp(377,17): error C4189: 'waitRet': local
       variable is initialized but not referenced [C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.v
       cxproj]
         C:\Users\micha\source\repos\terminal\src\tools\closetest\closetest.cpp(486,16): error C4189: 'success': local
       variable is initialized but not referenced [C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.v
       cxproj]
         C:\Users\micha\source\repos\terminal\src\tools\closetest\closetest.cpp(606,20): error C4189: 'success': local
       variable is initialized but not referenced [C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.v
       cxproj]

    10 Warning(s)
    4 Error(s)

Time Elapsed 00:12:25.35



### Expected Behavior

No warnings, no build errors.

### Actual Behavior

Build FAILED.

       "C:\Users\micha\source\repos\terminal\OpenConsole.sln" (Clean;Build target) (1) ->
       "C:\Users\micha\source\repos\terminal\src\host\ut_lib\host.unittest.vcxproj.metaproj" (Clean;Build target) (9) -
       >
       "C:\Users\micha\source\repos\terminal\src\host\proxy\Host.Proxy.vcxproj" (default target) (15:11) ->
       (Link target) ->
         LINK : warning LNK4266: missing load config symbol for image built with /GUARD [C:\Users\micha\source\repos\te
       rminal\src\host\proxy\Host.Proxy.vcxproj]


       "C:\Users\micha\source\repos\terminal\OpenConsole.sln" (Clean;Build target) (1) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\CascadiaPackage\CascadiaPackage.wapproj.metaproj" (Clean;Buil
       d target) (2) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\CascadiaPackage\CascadiaPackage.wapproj" (default target) (74
       :2) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\WindowsTerminal\WindowsTerminal.vcxproj" (GetTargetPath targe
       t) (76:5) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\Remoting\dll\Microsoft.Terminal.Remoting.vcxproj" (default ta
       rget) (75:5) ->
         LINK : warning LNK4001: no object files specified; libraries used [C:\Users\micha\source\repos\terminal\src\ca
       scadia\Remoting\dll\Microsoft.Terminal.Remoting.vcxproj]


       "C:\Users\micha\source\repos\terminal\OpenConsole.sln" (Clean;Build target) (1) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\CascadiaPackage\CascadiaPackage.wapproj.metaproj" (Clean;Buil
       d target) (2) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\CascadiaPackage\CascadiaPackage.wapproj" (default target) (74
       :2) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\WindowsTerminal\WindowsTerminal.vcxproj" (GetTargetPath targe
       t) (76:5) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsModel\dll\Microsoft.Terminal.Settings.Model.v
       cxproj" (default target) (84:9) ->
         LINK : warning LNK4001: no object files specified; libraries used [C:\Users\micha\source\repos\terminal\src\ca
       scadia\TerminalSettingsModel\dll\Microsoft.Terminal.Settings.Model.vcxproj]


       "C:\Users\micha\source\repos\terminal\OpenConsole.sln" (Clean;Build target) (1) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\CascadiaPackage\CascadiaPackage.wapproj.metaproj" (Clean;Buil
       d target) (2) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\CascadiaPackage\CascadiaPackage.wapproj" (default target) (74
       :2) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\WindowsTerminal\WindowsTerminal.vcxproj" (GetTargetPath targe
       t) (76:5) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.Settings.Editor.vcx
       proj" (default target) (87:7) ->
       (MarkupCompilePass2 target) ->
         C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\AddProfile.xaml(63,56): XamlCompiler
       warning WMC1506: OneWay bindings require at least one of their steps to support raising notifications when their
        value changes [C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.Sett
       ings.Editor.vcxproj]
         C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\ColorSchemes.xaml(119,44): XamlCompil
       er warning WMC1506: OneWay bindings require at least one of their steps to support raising notifications when th
       eir value changes [C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.S
       ettings.Editor.vcxproj]
         C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Interaction.xaml(58,37): XamlCompiler
        warning WMC1506: OneWay bindings require at least one of their steps to support raising notifications when thei
       r value changes [C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.Set
       tings.Editor.vcxproj]
         C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Launch.xaml(107,56): XamlCompiler war
       ning WMC1506: OneWay bindings require at least one of their steps to support raising notifications when their va
       lue changes [C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.Setting
       s.Editor.vcxproj]
         C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Launch.xaml(60,56): XamlCompiler warn
       ing WMC1506: OneWay bindings require at least one of their steps to support raising notifications when their val
       ue changes [C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.Settings
       .Editor.vcxproj]
         C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Appearances.xaml(56,40): XamlCompiler
        warning WMC1506: OneWay bindings require at least one of their steps to support raising notifications when thei
       r value changes [C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.Set
       tings.Editor.vcxproj]


       "C:\Users\micha\source\repos\terminal\OpenConsole.sln" (Clean;Build target) (1) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\LocalTests_TerminalApp\TerminalApp.LocalTests.vcxproj.metapro
       j" (Clean;Build target) (40) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\LocalTests_TerminalApp\TerminalApp.LocalTests.vcxproj" (defau
       lt target) (94:2) ->
       (Link target) ->
         TerminalAppLib.lib(XamlTypeInfo.g.obj) : warning LNK4197: export 'DllGetActivationFactory' specified multiple
       times; using first specification [C:\Users\micha\source\repos\terminal\src\cascadia\LocalTests_TerminalApp\Termi
       nalApp.LocalTests.vcxproj]


       "C:\Users\micha\source\repos\terminal\OpenConsole.sln" (Clean;Build target) (1) ->
       "C:\Users\micha\source\repos\terminal\src\host\ft_uia\Host.Tests.UIA.csproj.metaproj" (Clean;Build target) (18)
       ->
       "C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.vcxproj" (default target) (42:3) ->
       (ClCompile target) ->
         C:\Users\micha\source\repos\terminal\src\tools\closetest\closetest.cpp(301,17): error C4189: 'count2': local v
       ariable is initialized but not referenced [C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.vc
       xproj]
         C:\Users\micha\source\repos\terminal\src\tools\closetest\closetest.cpp(377,17): error C4189: 'waitRet': local
       variable is initialized but not referenced [C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.v
       cxproj]
         C:\Users\micha\source\repos\terminal\src\tools\closetest\closetest.cpp(486,16): error C4189: 'success': local
       variable is initialized but not referenced [C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.v
       cxproj]
         C:\Users\micha\source\repos\terminal\src\tools\closetest\closetest.cpp(606,20): error C4189: 'success': local
       variable is initialized but not referenced [C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.v
       cxproj]

    10 Warning(s)
    4 Error(s)

Time Elapsed 00:12:25.35

C:\Users\micha\source\repos\terminal>git show

--- a/src/tools/closetest/closetest.cpp
+++ b/src/tools/closetest/closetest.cpp
@@ -299,6 +299,7 @@ static std::vector<DWORD> getConsoleProcessList()
     assert(count1 >= 1 && "GetConsoleProcessList failed");
     ret.resize(count1);
     const DWORD count2 = GetConsoleProcessList(&ret[0], (DWORD)ret.size());
+    UNREFERENCED_PARAMETER(count2);
     assert(count1 == count2 && "GetConsoleProcessList failed");
     return ret;
 }
@@ -375,6 +376,7 @@ static void spawnChildTree(DWORD masterPid, const std::vector<std::wstring>& ext
     assert(success && "CreateProcessW failed");
 
     const DWORD waitRet = WaitForSingleObject(readyEvent, INFINITE);
+    UNREFERENCED_PARAMETER(waitRet);
     assert(waitRet == WAIT_OBJECT_0 && "WaitForSingleObject failed");
     CloseHandle(readyEvent);
     CloseHandle(pi.hThread);
@@ -485,6 +487,7 @@ static HANDLE duplicateHandle(HANDLE srcProc, HANDLE srcHandle)
     HANDLE ret{};
     const auto success =
         DuplicateHandle(srcProc, srcHandle, GetCurrentProcess(), &ret, 0, FALSE, DUPLICATE_SAME_ACCESS);
+    UNREFERENCED_PARAMETER(success);
     assert(success && "DuplicateHandle failed");
     return ret;
 }
@@ -605,6 +608,7 @@ static int doChild(std::deque<std::wstring> argv)
     {
         const BOOL success =
             AssignProcessToJobObject(jobHandle, GetCurrentProcess());
+        UNREFERENCED_PARAMETER(success);
         assert(success && "AssignProcessToJobObject failed");
         CloseHandle(jobHandle);
         jobHandle = nullptr;

